src/pages/VaultsPage/WithdrawModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState, useEffect } from 'react';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { GraphQLTypes } from 'lib/gql/__generated__/zeus';
+import { Contracts } from 'lib/vaults';
+import { useForm, useController } from 'react-hook-form';
+import * as z from 'zod';
+
+import { FormTokenField } from 'components';
+import { useContracts } from 'hooks/useContracts';
+import { useVaultRouter } from 'hooks/useVaultRouter';
+import { Form, Button, Modal } from 'ui';
+
+export type WithdrawModalProps = {
+  onClose: () => void;
+  onWithdraw: () => void;
+  vault: GraphQLTypes['vaults'];
+  balance: number;
+};
+export default function WithdrawModal({
+  onClose,
+  onWithdraw,
+  vault,
+  balance,
+}: WithdrawModalProps) {
+  const schema = z.object({ amount: z.number().min(0).max(balance) }).strict();
+  type WithdrawFormSchema = z.infer<typeof schema>;
+  const contracts = useContracts();
+  const [submitting, setSubmitting] = useState(false);
+  const [selectedContracts, setSelectedContracts] = useState<Contracts>();
```

Comment:
let's remove `selectedContracts` and just pass `contracts` directly to `useVaultRouter`, since withdrawing to an arbitrary address isn't supported by the contracts